app/assets/javascripts/Components/Modals/extension_modal.jsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import Modal from 'react-modal';
+
+
+class ExtensionModal extends React.Component {
+
+  static defaultProps = {
+    weeks: 0,
+    days: 0,
+    hours: 0,
+    note: '',
+    penalty: false,
+    updating: false
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      weeks: props.weeks,
+      days: props.days,
+      hours: props.hours,
+      note: props.note,
+      penalty: props.penalty
+    }
```

Comment:
You have a few statements throughout that are missing semicolons. Please add them all in.